### PR TITLE
publish node components version information

### LINKED ISSUE #5948: Kubelet publish node components' version
Forked from #4855, and required for rolling upgrade: 

Kubelet should report cross-products version, such kernel, docker, kubelet, kube-proxy, image, etc. to API server. 

cc/ @zmerlynn

### PR DESCRIPTION
**Pull request by dchen1107**
With this PR, kubectl describe nodes id includes related version info.

Running: cluster/../cluster/gce/../../cluster/../_output/dockerized/bin/linux/amd64/kubectl describe nodes kubernetes-minion-4vox.c.golden-system-455.internal
Name:           kubernetes-minion-4vox.c.golden-system-455.internal
Labels:         <none>
CreationTimestamp:  Mon, 30 Mar 2015 16:20:29 -0700
Conditions:
  Type      Status  LastProbeTime               LastTransitionTime          Reason                  Message
  Ready     True    Tue, 31 Mar 2015 00:35:37 -0700     Mon, 30 Mar 2015 16:20:59 -0700     kubelet is posting ready status  
Addresses:  130.211.182.208,104.197.28.200
Capacity:
 cpu:       1
 memory:    3800824Ki
Version:
 Kernel Version:        3.16.0-0.bpo.4-amd64
 OS Image:          Debian GNU/Linux 7 (wheezy)
 Container Runtime Version: docker://1.5.0
 Kubelet Version:       v0.14.0-10-g3e0cdff97c2e77-dirty
 Kube-Proxy Version:        v0.14.0-10-g3e0cdff97c2e77-dirty
ExternalID:         9471772650125516840
Pods:               (2 in total)
  fluentd-to-elasticsearch-kubernetes-minion-4vox
  monitoring-influx-grafana-controller-240nz
Events:
  FirstSeen             LastSeen            Count   From                                SubobjectPath   Reason      Message
  Tue, 31 Mar 2015 00:17:25 -0700   Tue, 31 Mar 2015 00:17:25 -0700 1   {kubelet kubernetes-minion-4vox.c.golden-system-455.internal}           starting    Starting kubelet.

Fix #5948

### CODE REVIEW

File: pkg/kubelet/kubelet.go [hunk 1]
```diff
@@ -2075,3 +2089,9 @@ func (kl *Kubelet) GetCachedMachineInfo() (*cadvisorApi.MachineInfo, error) {
 	}
 	return kl.machineInfo, nil
 }
+
+// GetNodeVersionInfo returns version information (from Cadvisor) of current machine
+func (kl *Kubelet) GetNodeVersionInfo() (*cadvisorApi.VersionInfo, error) {
```
Comment (vmarmol) [inline pkg/kubelet/kubelet.go]: I think we don't need this function, we can probably call cAdvisor directly above. We only export the information through status anyways.

**Comment by vmarmol** [comment]
Thanks @dchen1107! Minor comment, LGTM otherwise.

File: pkg/kubelet/kubelet.go (hunk 1 above)
Comment (dchen1107) [inline pkg/kubelet/kubelet.go]: Yeah, removed it. Thanks!

**Comment by vmarmol** [comment]
LGTM, we can squash the last commit and merge on green :)

**Comment by dchen1107** [comment]
Done with squash!

**Pull request merged**